src/main/java/graphs/GraphEdge.java

```diff
@@ -96,13 +67,11 @@ public GraphNode getOtherNode(GraphNode node) {
 
     // "Note: this class has a natural ordering that is inconsistent with equals."
     public int compareTo(GraphEdge edge) {
-        if (isWeighted && edge.isWeighted()) {
-            // is it necessary to specify "this." ?
-            if(edge.getWeight() > this.getWeight()) {
-                return -1;
-            } else if (this.getWeight() > edge.getWeight()){
-                return 1;
-            }
+        // is it necessary to specify "this." ?
+        if(edge.getWeight() > this.getWeight()) {
```

Comment:
so the compare to method just cares about less than, greater than, or equal to zero. it doesnt need to handle the margin of difference. since im just returning 0, -1 or 1, I could change it to Double compareTo(), but it looks like the Comparable interface says the method should return an int, if i change that would the Comparable interface not be implemented correctly and thus Collection.sort() wouldnt work?